worlds/sm64ex/Rules.py

```diff
@@ -1,33 +1,54 @@
-from ..generic.Rules import add_rule
-from .Regions import connect_regions, sm64_level_to_paintings, sm64_paintings_to_level, sm64_level_to_secrets, sm64_entrances_to_level, sm64_level_to_entrances
+from typing import Callable, Union
 
-def shuffle_dict_keys(world, obj: dict) -> dict:
-    keys = list(obj.keys())
-    values = list(obj.values())
+from BaseClasses import MultiWorld
+from ..generic.Rules import add_rule, set_rule
+from .Locations import location_table
+from .Regions import connect_regions, sm64_level_to_paintings, sm64_paintings_to_level, sm64_level_to_secrets, \
+sm64_level_to_secrets, sm64_entrances_to_level, sm64_level_to_entrances
+from .Items import action_item_table
+
+def shuffle_dict_keys(world, dictionary: dict) -> dict:
+    keys = list(dictionary.keys())
+    values = list(dictionary.values())
     world.random.shuffle(keys)
-    return dict(zip(keys,values))
+    return dict(zip(keys, values))
 
-def fix_reg(entrance_ids, entrance, destination, swapdict, world):
-    if entrance_ids[entrance] == destination: # Unlucky :C
+def fix_reg(entrance_map, entrance, destination, swapdict, world):
+    if entrance_map[swapdict[entrance]] == destination: # Unlucky :C
```

Comment:
Should probably be `entrance_map[sm64_entrances_to_level[entrance]]` right?

Reply:
I'm having some trouble figuring out how swap_dict works, and it doesn't seem to be working correctly before or after that change.  I'm going to have to take a look at it later; might make an enum for Entrances to make it easier to understand.